src/pages/iou/IOUTransactions.js

```diff
@@ -27,6 +28,37 @@ const propTypes = {
     /** Does the associaed have an outstanding IOU? */
     hasOutstandingIOU: PropTypes.bool,
 
+    /** IOU Report data object */
+    iouReport: PropTypes.shape({
+        /** The report ID of the IOU */
+        reportID: PropTypes.number,
+
+        /** The report ID of the chat associated with the IOU */
+        chatReportID: PropTypes.number,
+
+        /** The total amount in cents */
+        total: PropTypes.number,
+
+        /** The owner of the IOUReport */
+        ownerEmail: PropTypes.string,
+
+        /** The currency of the IOUReport */
+        currency: PropTypes.string,
+    }).isRequired,
+
+    /** All of the personal details for everyone */
+    personalDetails: PropTypes.objectOf(PropTypes.shape({
+
+        /** This is either the user's full name, or their login if full name is an empty string */
+        displayName: PropTypes.string.isRequired,
+    })).isRequired,
+
+
+    /** Session of currently logged in user */
+    session: PropTypes.shape({
```

Comment:
NAB but `session` is used pretty widely by many different components + we could create an Onyx context for it.

Reply:
I've added `withSession` provider for my changes at least. I can see 29 other uses of `session: { key: ONYXKEYS.SESSION }`.  What should we do about those?

Reply:
@marcaaron Can you buddy-check this? We've created a `withSession` provider, and I just wanted to make sure that there wasn't some reason you didn't do that already, or if you have an opinion about whether we should include this change / refactor existing uses of the session key relying on `withOnyx`.

Reply:
Alright, this should probably be better documented, but there's a really simple rule for when to create new context..

>Is the component that needs to subscribe to an Onyx key something that could theoretically show up 100s of times in a list? If so, then you should use context.

Each one of those items would create a separate Onyx connection which we've learned is really bad for performance.

In this case,  I'd say we don't need to use context at all because...

1. There will only ever be one IOUAction rendered
2. The IOU Transactions page is just rendered once

If my assumptions about when those things will render are wrong then I would change my thoughts.

Reply:
`IOUAction` would be rendered multiple times based on the number of transactions messages in the chat list.

Take the following snapshot, `IOUAction` is rendered thrice for 3 different messages:

<img width="791" alt="Screenshot 2021-08-24 at 11 01 53 PM" src="https://user-images.githubusercontent.com/3069065/130664291-259aa3da-3ee7-4aa3-920e-7514dcc84d47.png">

Reply:
Hmm, got it, thanks. In that case, I'd still maybe guess there's not much value in using context here. But we really need to benchmark to see if there's much difference - which feels out of scope for this PR/issue.

I'd think most chats would not be very long lists of IOU actions. So it feels like we are optimizing for an edge case and can probably start by doing the simpler thing of just using `withOnyx` for now. I'd much prefer that we use the context strategy in a reactive way when there is some clear performance issue we need to solve vs. pulling for it with the assumption that it is always better to use.

Reply:
Okay got it. Thanks for the detailed explanation.